setup.py

```diff
@@ -22,10 +22,140 @@
 #
 
 from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext
+import sysconfig
+import platform
 
 with open("README.md", "r") as file_long_description:
     long_description = file_long_description.read()
 
+BUILD_DIR = "build"
+machine = platform.machine().lower()  # TODO: this is unreliable, need something better to detect target architecture
+is_armv7 = (machine.startswith("armv7"))  # TODO: can armv8_ still be AArch32?
```

Comment:
My NAS has a  ARMv8  and `platform.machine` reports `aarch64`.
So it doesn't include the NEON stuff?

Reply:
AArch64 applies NEON by default (and `-mfpu=neon` isn't recognised).

Reply:
Check, good to know.. We will probably have to introduce some support for command line parameters to specify custom target arch so we can support cross compiling.
Have go check how nzbget fixed this in his universal Linux installer.

Reply:
Getting the arch is dependent on the build system, so you can't really take a solution from a different build system and use it here.  
Ideally, CCompiler would have some sort of 'get_arch' function, as the compiler would know what it's targeting.

It doesn't look like there's much support for cross compiling in setuptools. It seems like it has some support for MSVC, but otherwise, examples I've seen seem to mostly override the selected compiler with a cross compiler, rather than use something built into setuptools.

I did come across `distutils.util.get_platform()`, which looks better, but don't know how definitive it is.

Overall, from my quick look, setuptools doesn't look like an extensive build system, so you may have to live with its limitations.